src/preferences/dialog/dlgprefeffects.cpp

```diff
@@ -62,22 +116,21 @@ void DlgPrefEffects::slotUpdate() {
 }
 
 void DlgPrefEffects::slotApply() {
-    const QList<EffectProfilePtr> effectProfiles = m_availableEffectsModel.profiles();
-    for (const EffectProfilePtr& profile : effectProfiles) {
-        EffectManifestPointer pManifest = profile->pManifest;
-        m_pEffectsManager->setEffectVisibility(pManifest, profile->bIsVisible);
-
-        // Effects from different backends can have same Effect IDs.
-        // Add backend name to group to uniquely identify those effects.
-        // Use untranslated value to keep the group language independent.
-        m_pConfig->set(ConfigKey("[Visible " + pManifest->backendName() + " Effects]", pManifest->id()),
-                ConfigValue(profile->bIsVisible));
-    }
+    m_pVisibleEffectsList->setList(m_pVisibleEffectsModel->getList());
+    saveChainPresetLists();
```

Comment:
Changing this would require adding a lot of complexity which I'm not going to do in this PR.

Reply:
Really? Does this statement still apply after you implement the reset to default button? Isn't this the same thing? Can't we just make a local copy of the data structures? 

It was working in the old Version. So it is a regression.

Reply:
Added to the 2.4 project: https://github.com/mixxxdj/mixxx/projects/3#card-64300860

> It was working in the old Version. So it is a regression.

This does not make sense. This is a new feature.

Reply:
Please also add a FIXME comment to an appropriate entry point of the code with a link to the card. Bidirectional linking is handy.

All new editing features should be designed by working on a copy instead of the modifying the current state directly. But I agree that we should resolve this issue separately after merging.